CMakeLists.txt

```diff
@@ -100,8 +108,22 @@ FetchContent_MakeAvailable(repo-common repo-core repo-backend)
 #
 if(${TRITON_ENABLE_GPU})
   find_package(CUDAToolkit REQUIRED)
+else()
```

Comment:
Instead of turning it off you should throw an error
See how to do this in the ORT backend: https://github.com/triton-inference-server/onnxruntime_backend/blob/main/CMakeLists.txt#L101-L105

Reply:
That would not work right if we set the default to ON, would it ?

Reply:
Why do you think it would not work right? The GPU is enabled by default as well. If the user disabled GPU support, we should require him to turn off Torch TRT support as well.

Reply:
I think it's a burden on the user - if he uses ON default, to require also specifically  add TRITON_PYTORCH_ENABLE_TORCHTRT=OFF - an option he obviously does not care about. And I can't tell if the user specified it to be on or have used the default.

Reply:
Look at the new logic - most of those diffs are outdated.

Reply:
Now the logic is : enabled by default, and we do checks if it's actually available, and turn off if it's not. Also, user can explicitly turn it OFF.